// angular2-busy/build/busy-config.js
"use strict";
/**
 * @file Busy Config
 * @author yumao<yuzhang.lille@gmail.com>
 */
exports.__esModule = true;
var BusyConfig = /** @class */ (function () {
    function BusyConfig(config) {
        if (config === void 0) { config = {}; }
        for (var option in exports.BUSY_CONFIG_DEFAULTS) {
            this[option] = config[option] != null ? config[option] : exports.BUSY_CONFIG_DEFAULTS[option];
        }
    }
    return BusyConfig;
}());
exports.BusyConfig = BusyConfig;
exports.BUSY_CONFIG_DEFAULTS = {
    template: "\n        <div class=\"ng-busy-default-wrapper\">\n            <div class=\"ng-busy-default-sign\">\n                <div class=\"spinner\">\n                  <div class=\"rect1\"></div>\n                  <div class=\"rect2\"></div>\n                  <div class=\"rect3\"></div>\n                  <div class=\"rect4\"></div>\n                  <div class=\"rect5\"></div>\n            </div>\n            <div class=\"ng-busy-default-text\">{{message}}</div>\n            </div>\n        </div>\n    ",
    delay: 0,
    minDuration: 0,
    backdrop: true,
    message: 'Please wait...',
    wrapperClass: 'ng-busy'
};
